### ISSUE TITLE
email-list stores emails as string instead of appending array

### ISSUE THREAD

**Issue by synack-security** [cloud-custodian/cloud-custodian#5228]
Using "account_emails" in c7n-mailer mailer.yml config file doesn't work because the get_valid_emails_from_list function is expecting an array, but is getting a string.

Line 134 of email_delivery.py stores email as string in "email_list" instead of appending the email to the array "email_list".

Change from:
email_list = account_email_mapping.get(account_id, [])
Change To:
email_list.append(account_email_mapping.get(account_id, []))

**Comment by kapilt** [comment]
hmm.. this attribute looks under specified in the config schema, we probably should be more specific that its an array in schema, and potentially support both string or array.

simply changing this from array to a string, would break existing configurations?

**Comment by synack-security** [comment]
> hmm.. this attribute looks under specified in the config schema, we probably should be more specific that its an array in schema, and potentially support both string or array.
> 
> simply changing this from array to a string, would break existing configurations?

Agreed.  I was able to modify my mailer.yml and make my config compatible as an array.  The documentation should reflect that an array is required currently, but enhancement would be to allow a single value/string (ie 123455666: email@thing.com).
